test/html/normalize.test.js

```diff
@@ -1,231 +1,74 @@
-		)
-
-		var dom = new DOMParser().parseFromString(
-			'<div><123e>&<a<br/></div>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<div xmlns="http://www.w3.org/1999/xhtml">&lt;123e>&amp;&lt;a<br/></div>'
-		)
-
-		var dom = new DOMParser().parseFromString(
-			'<div>&nbsp;&copy;&nbsp&copy</div>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<div xmlns="http://www.w3.org/1999/xhtml">\u00a0\u00a9&amp;nbsp&amp;copy</div>'
-		)
-
-		var dom = new DOMParser().parseFromString(
-			'<html xmlns:x="1"><body/></html>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<html xmlns:x="1" xmlns="http://www.w3.org/1999/xhtml"><body></body></html>'
-		)
+	it.each([
+		'<div>&amp;&lt;123&456<789;&&</div>',
+		'<div>&amp;&lt;123&456<789;&&</div>',
+		'<div><123e>&<a<br/></div>',
+		'<div>&nbsp;&copy;&nbsp&copy</div>',
+		'<html xmlns:x="1"><body/></html>',
+		'<html test="a<b && a>b && \'&amp;&&\'"/>',
+		'<div test="alert(\'<br/>\')"/>',
+		'<div test="a<b&&a< c && a>d"></div>',
+		'<div a=& bb c d=123&&456/>',
+		'<div a=& a="&\'\'" b/>',
+		'<html test="123"/>',
+		'<r><Label onClick="doClick..>Hello, World</Label></r>',
+		'<Label onClick=doClick..">Hello, World</Label>',
+	])('%s', (xml) => {
+		const { errors, parser } = getTestParser()
+
+		const actual = parser.parseFromString(xml, 'text/html').toString()
+
+		expect({ actual, ...errors }).toMatchSnapshot()
 	})
 
-	it('attr', () => {
-		var dom = new DOMParser().parseFromString(
-			'<html test="a<b && a>b && \'&amp;&&\'"/>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<html test="a&lt;b &amp;&amp; a>b &amp;&amp; \'&amp;&amp;&amp;\'" xmlns="http://www.w3.org/1999/xhtml"></html>'
-		)
+	it.each([
+		'<html><meta><link><img><br><hr><input></html>',
+		'<html title =1/2></html>',
+		'<html title= 1/>',
+		'<html title = 1/>',
+		'<html title/>',
+		'<html><meta><link><img><br><hr><input></html>',
+	])('unclosed html %s', (xml) => {
+		const { errors, parser } = getTestParser()
 
-		var dom = new DOMParser().parseFromString(
-			'<div test="alert(\'<br/>\')"/>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<div test="alert(\'&lt;br/>\')" xmlns="http://www.w3.org/1999/xhtml"></div>'
-		)
-		var dom = new DOMParser().parseFromString(
-			'<div test="a<b&&a< c && a>d"></div>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<div test="a&lt;b&amp;&amp;a&lt; c &amp;&amp; a>d" xmlns="http://www.w3.org/1999/xhtml"></div>'
-		)
+		const actual = parser.parseFromString(xml, 'text/html').toString()
 
-		var dom = new DOMParser().parseFromString(
-			'<div a=& bb c d=123&&456/>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<div a="&amp;" bb="bb" c="c" d="123&amp;&amp;456" xmlns="http://www.w3.org/1999/xhtml"></div>'
-		)
-
-		var dom = new DOMParser().parseFromString(
-			'<div a=& a="&\'\'" b/>',
-			'text/html'
-		)
-		assert(
-			dom + '',
-			'<div a="&amp;\'\'" b="b" xmlns="http://www.w3.org/1999/xhtml"></div>'
-		)
+		expect({ actual, ...errors }).toMatchSnapshot()
```

Comment:
If you look at the snapshots that's what is happening. If at any point in the future an error is raised for any of those the test will fail. And since the `errors` dict is spread, there is no trace of it in the snapshot.